registry/storage/driver/s3-aws/s3.go

```diff
@@ -1389,8 +1332,7 @@ func (d *driver) newWriter(ctx context.Context, key, uploadID string, parts []*s
 		uploadID: uploadID,
 		parts:    parts,
 		size:     size,
-		ready:    d.NewBuffer(),
-		pending:  d.NewBuffer(),
+		buf:      d.pool.Get().(*bytes.Buffer),
 	}
 }
 
```

Comment:
Question, unrelated to the changes here (sorry if distracting): Why/how could the parts ever become unordered? Shouldn't it be sufficient to sort the parts once when they are first fetched rather than when the upload is restarted or committed?

Reply:
I think I'm missing some context for this 🤔